services/motion/builtin/builtin.go

```diff
@@ -410,11 +410,21 @@ func (ms *builtIn) planMoveOnMap(
 		}
 	}
 
+	fs, err := ms.fsService.FrameSystem(ctx, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
 	kb, err := kinematicbase.WrapWithKinematics(ctx, b, ms.logger, motion.NewSLAMLocalizer(slamSvc), limits, kinematicsOptions)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// replace original base frame with kb frame
```

Comment:
Can we put a couple lines of comment here saying why a replacement is necessary?